rmgpy/rmg/main.py

```diff
@@ -1741,6 +1746,11 @@ def save_everything(self):
         # Notify registered listeners:
         self.notify()
 
+        if self.profiler:  # Save the profile information in case the job crashes
+            with open(self.stats_file, 'wb') as f:
+                self.profiler.snapshot_stats()
+                marshal.dump(self.profiler.stats, f)
```

Comment:
It seems like profile saving time will be included in the profile. Is this worth trying to avoid?

Reply:
Absolutely, if possible this would be ideal. However, I think this would only be possible if we call `Profile.disable()` or equivalent, which seemed to cause other errors.

I don't remember seeing saving the profile information in the profile graph, I'll look to see how much time it is taken. presumably this is negligible.

Reply:
Quick follow-up: Saving the profile information every iteration does show up on the graph. On the average this takes 0.1 seconds per iteration based on running the minimal example. For the minimal example this amounted to 2% of the total run time, but I would expect for actual profiling jobs this will be negligible.

Reply:
I was actually wondering if it will become more substantial for longer jobs as the amount of data accumulates.

Reply:
I can test this out, but my hunch is that it will not. Unlike the chemkin file, which accumulates species and reactions each iteration, the profile file should not accumulate new functions (all of the relevant functions should be called at least once after the first iteration), it should just be updating the values of int/floats used to store the times and number of calls for each function. This at least is my naive belief about how the code should work, but it is worth testing

Reply:
I made a separate function for saving the time profiling information so cProfile could explicitly time that. For superminimal, saving the time profiling info took 6.12% of the total job time at ~0.06s per iteration. For minimal, saving the time profiling info took just 2.18% of the total job time at ~0.08s per iteration. Since the percentage of total time spent saving the info decreases as the job size increases, I agree with @amarkpayne that it should scale well with large jobs. @mliu49 I think this branch is ready to be merged 👍